main.go

```diff
@@ -17,6 +23,95 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+type MigrationRunner struct {
+	path string
+}
+
+func NewMigrationRunner(path string) *MigrationRunner {
+	return &MigrationRunner{path: path}
+}
+
+func (m *MigrationRunner) Run(ctx context.Context, conn *sql.Conn, lastRanId int) error {
+	files, err := m.loadMigrationFiles()
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(files, func(i, j int) bool {
+		id1 := extractMigrationID(files[i])
+		id2 := extractMigrationID(files[j])
+
+		return id1 < id2
+	})
+
+	// var index int = 0
+	// for _, file := range files {
+	// 	if extractMigrationID(file) <= lastRanId {
+	// 		index++
+	// 	}
+	// }
+
+	fileSlice := files[lastRanId+1:]
+
+	for _, file := range fileSlice {
+		readFile, err := os.Open(file)
+
+		if err != nil {
+			fmt.Println(err)
+		}
+		fileScanner := bufio.NewScanner(readFile)
+
+		fileScanner.Split(bufio.ScanLines)
+		for fileScanner.Scan() {
+
+			if strings.Trim(fileScanner.Text(), "") == "" {
+				continue
+			}
+			if _, err := conn.ExecContext(ctx, fileScanner.Text()); err != nil {
+				return fmt.Errorf("failed to execute migration %s: %w", file, err)
+			}
+		}
+
+		readFile.Close()
```

Comment:
You typically would want to put this in a `defer` statement right after the file was opened:
```go
readFile, err := os.Open(file)
if err != nil {
	return err 
}
defer readFile.Close()
```

That way we don't exit the function without closing the file first (which would happen if/when we encountered an error).